lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -28,16 +28,52 @@ class KubernetesResource
 
     TIMEOUT_OVERRIDE_ANNOTATION = "kubernetes-deploy.shopify.io/timeout-override"
 
-    def self.build(namespace:, context:, definition:, logger:)
-      opts = { namespace: namespace, context: context, definition: definition, logger: logger }
-      if KubernetesDeploy.const_defined?(definition["kind"])
-        klass = KubernetesDeploy.const_get(definition["kind"])
-        klass.new(**opts)
-      else
-        inst = new(**opts)
-        inst.type = definition["kind"]
-        inst
-      end
+    def self.inherited(child_class)
+      KubernetesResource.child_classes.add(child_class)
+    end
+
+    def self.child_classes
+      @child_classes ||= Set.new
+    end
+
+    def self.all
+      child_classes.dup
+    end
+
+    def self.prunable?
+      self::PRUNABLE if defined? self::PRUNABLE
```

Comment:
nit: these should be `false` rather than `nil` when the constants aren't available